pandas/core/indexes/base.py

```diff
@@ -357,7 +354,7 @@ class Index(IndexOpsMixin, PandasObject):
     #  given the dtypes of the passed arguments
 
     @final
-    def _left_indexer_unique(self: _IndexT, other: _IndexT) -> npt.NDArray[np.intp]:
+    def _left_indexer_unique(self, other: Index) -> npt.NDArray[np.intp]:
```

Comment:
Is this change intended? Is `_left_indexer_unique(self, other: Self)` to strict?

Reply:
(same for the following methods)

Reply:
> For the built-in index classes I think it will always require same class (so `other: Self`)

I would then stick with `Self` it is also stricter which could help catch bugs (or causes a false mypy error which can then be fixed by someone who knows for sure).